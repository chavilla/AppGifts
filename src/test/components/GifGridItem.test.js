import React from 'react';
import { shallow } from "enzyme";
import GifGridItem from '../../components/GifGridItem';

describe('Test in <GifGridItem />', () => {
    

    const title='Hola';

    const url='www.google.com';

    const wrapper=shallow(<GifGridItem title={title} url={url} />)

    test('show GifGridItem component',()=>{

        expect(wrapper).toMatchSnapshot();

    });

    test('should have a paragraph with title',()=>{
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('should have an img equal url',()=>{
        const img = wrapper.find('img');

        expect(img.props().src).toBe(url);
    });

    test('verify that the div has a className animate__fadeIn',()=>{
        const classNameAnimate= 'animate__fadeIn';

        const div=wrapper.find('div');
 
        expect( div.props().className.includes(classNameAnimate)).toBe(true)

    });

})