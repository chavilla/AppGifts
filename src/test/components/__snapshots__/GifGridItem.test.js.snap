// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> show GifGridItem component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola"
    src="www.google.com"
  />
  <p>
    Hola
  </p>
</div>
`;
